### PR TITLE
[wip] Ensure one PTR for the same IP

### PR DESCRIPTION
**Pull request by ghost**
Before: Creation of two Records with the same IP ended up with two PTRs.
Now: Creation of two Records with the same IP ends up with only one PTR
for the first Record.

### CODE REVIEW

File: powerdns/tests/test_auto_ptr.py [hunk 1]
```diff
@@ -73,6 +73,23 @@ def test_default_ptr_created(self):
...
+            auto_ptr=AutoPtrOptions.ALWAYS,
+        )
+        RecordFactory(
+            domain=self.domain,
+            type='A',
+            name='ptr-reuse.example.com',
+            content=record_creating_ptr.content,
+            auto_ptr=AutoPtrOptions.ALWAYS,
+        )
+        self.assertEqual(Record.objects.filter(type='PTR').count(), 1)
```
Comment (mkurek) [inline powerdns/tests/test_auto_ptr.py:91]: check here also for PTR content

File: powerdns/models/powerdns.py [hunk 2]
```diff
@@ -591,17 +595,21 @@ def create_ptr(self):
...
-            domain=domain,
-            name='.'.join([number, domain_name]),
-            content=self.name,
-            depends_on=self,
-            owner=self.owner,
-            ttl=self.ttl,
-            disabled=self.disabled,
-        )
+        # ensure 1 PTR (at the most) exist for the same IP
+        if not Record.objects.filter(
```
Comment (mkurek) [inline powerdns/models/powerdns.py:602]: couldn't we use here https://docs.djangoproject.com/en/1.9/ref/models/querysets/#update-or-create ?

File: powerdns/tests/test_auto_ptr.py (hunk 1 above)
Comment (ghost) [inline powerdns/tests/test_auto_ptr.py:91]: done

File: powerdns/models/powerdns.py [hunk 3]
```diff
@@ -565,7 +565,14 @@ def save(self, *args, **kwargs):
         super(Record, self).save(*args, **kwargs)
 
     def delete_ptr(self):
```
Comment (mkurek) [inline powerdns/models/powerdns.py:567]: Add some comment what is self here (A record, isn't it?)

File: powerdns/models/powerdns.py [hunk 4]
```diff
@@ -565,7 +565,14 @@ def save(self, *args, **kwargs):
...
-        Record.objects.filter(depends_on=self).delete()
+        records = Record.objects.filter(type='A', content=self.content)
+        if records.count() == 1:
+            # no more A Record with this IP exists, delete PTR
+            Record.objects.filter(depends_on=self).delete()
```
Comment (mkurek) [inline powerdns/models/powerdns.py]: Why not `records.delete()`?

**Comment by ghost** [comment]
This needs update because concept has changed (see https://github.com/allegro/django-powerdns-dnssec/pull/125)

**Pull request closed without merging**